Sources/ClientRuntime/Networking/Http/HttpClientConfiguration.swift

```diff
@@ -34,6 +35,18 @@ public class HttpClientConfiguration {
     /// If none is provided, the default protocol for the operation will be used
     public var protocolType: ProtocolType?
 
+    /// Custom TLS configuration for HTTPS connections.
+    ///
+    /// Enables specifying client certificates and trust stores for secure communication.
+    /// Defaults to system's TLS settings if `nil`.
+    public var tlsContext: TLSContext?
+
+    /// Custom TLS configuration for HTTPS connections with URLSession client.
+    ///
+    /// Enables specifying client certificates and trust stores for secure communication.
+    /// Defaults to system's TLS settings if `nil`.
+    public var urlSessionTLSOptions: URLSessionTLSOptions?
```

Comment:
If this is the public API, Is it possible to have one TLSContext class and create either a URLSessionTLSOptions or a TLSContext internally from it?

Reply:
If this is a public API, it might be better to just expose a wrapper to TLSContextOptions and create a TLSContext from that. No one should be passing server mode to this TLSContext.init function.